lib/task-graph.js

```diff
@@ -24,6 +25,7 @@ function taskGraphFactory(
     Task,
     store,
     taskMessenger,
+    waterline,
```

Comment:
In the original architecture, we intentionally encapsulated all database queries within the `TaskGraph.Store` dependency, partly to help loosen our dependency on waterline, and partly to allow for database-specific query optimizations. I would very strongly suggest doing the same here.